src/main/scala/gw/cda/api/outputwriter/OutputWriter.scala

```diff
@@ -74,9 +171,724 @@ trait OutputWriter {
+
+    // Log total rows to be merged for this fingerprint.
+    val totCnt = tableDataFrameWrapperForMicroBatch.dataFrame.count()
+    log.info(s"Merged - $tableName total cnt for all ins/upd/del: ${totCnt.toString}")
+
+    // Filter for records to insert and drop unwanted columns.
+    val InsertDF = tableDataFrameWrapperForMicroBatch.dataFrame.filter(col("gwcbi___operation").isin(2, 0))
+      .drop(dropList: _*)
+      .cache()
+
+    // Log total rows to be inserted for this fingerprint.
+    val insCnt = InsertDF.count()
+    log.info(s"Merged - $tableName insert cnt after filter: ${insCnt.toString}")
+
+    // Determine if we need to create the table by checking if the table already exists.
+    val url = clientConfig.jdbcConnectionMerged.jdbcUrl
+    val dbm = connection.getMetaData
+    val dbProductName = dbm.getDatabaseProductName
+    val tableNameCaseSensitive = dbProductName match {
+      case "Microsoft SQL Server" => tableNameNoSchema
+      case "PostgreSQL"           => tableNameNoSchema
+      case "Oracle"               => tableNameNoSchema.toUpperCase
+      case _                      => throw new SQLException(s"Unsupported database platform: $dbProductName")
+    }
+    val tables = dbm.getTables(connection.getCatalog(), connection.getSchema(), tableNameCaseSensitive, Array("TABLE"))
+    val tableExists = tables.next
+
+    // Get some data we will need for later.
+    val dialect = JdbcDialects.get(url)
+    val insertSchema = InsertDF.schema
+    val batchSize = 5000 // TODO consider making this configurable.
+
+    // Create the table if it does not already exist.
+    if (!tableExists) {
+      // Build create table statement.
+      val createTableDDL = getTableCreateDDL(dialect, insertSchema, tableName, JdbcWriteType.Merged, dbProductName)
+      log.info(s"Merged - $createTableDDL")
+      // Execute the table create DDL
+      val stmt = connection.createStatement
+      stmt.execute(createTableDDL)
+      stmt.close()
+
+      // Create table indexes for the new table.
+      createIndexes(connection, url, tableName, JdbcWriteType.Merged)
+      connection.commit()
+    }
+
+    // Build the insert statement.
+    val columns = insertSchema.fields.map(x => dialect.quoteIdentifier(x.name)).mkString(",")
+    val placeholders = insertSchema.fields.map(_ => "?").mkString(",")
+    val insertStatement = s"INSERT INTO $tableName ($columns) VALUES ($placeholders)"
+    log.info(s"Merged - $insertStatement")
+
+    // Prepare and execute one insert statement per row in our insert dataframe.
+    updateDataframe(connection, tableName, InsertDF, insertSchema, insertStatement, batchSize, dialect, JdbcWriteType.Merged)
+
+    InsertDF.unpersist()
+
+    // Filter for records to update.
+    val UpdateDF = tableDataFrameWrapperForMicroBatch.dataFrame.filter(col("gwcbi___operation").isin(4))
+      .cache()
+
+    // Log total rows marked as updates.
+    val UpdCnt = UpdateDF.count()
```

Comment:
```suggestion
    val updateCount = UpdateDF.count()
```
Why the abbreviation?

Reply:
agreed. completed locally.